### ISSUE TITLE
Prometheus failed to start.

### ISSUE THREAD

**Issue by Kirchen99** [prometheus/prometheus#5727]
## Bug Report
**What did you do?**
Start Prometheus in Kubernetes.

**What did you expect to see?**
Prometheus UI

**What did you see instead? Under which circumstances?**
Service Unavailable
Prometheus Pod restarts without error message.
Exit Code is 137 and it is not OOM killed.

**Environment**

* System information:

	Kubernetes v1.12.7
* Prometheus version:

	v2.10

* Logs:
```
{"caller":"main.go:275","deprecation_notice":"'storage.tsdb.retention' flag is deprecated use 'storage.tsdb.retention.time' instead.","level":"warn","ts":"2019-07-02T08:35:21.997Z"}
{"caller":"main.go:322","level":"info","msg":"Starting Prometheus","ts":"2019-07-02T08:35:21.998Z","version":"(version=2.10.0, branch=HEAD, revision=d20e84d0fb64aff2f62a977adc8cfb656da4e286)"}
{"build_context":"(go=go1.12.5, user=root@a49185acd9b0, date=20190525-12:28:13)","caller":"main.go:323","level":"info","ts":"2019-07-02T08:35:21.998Z"}
{"caller":"main.go:324","host_details":"(Linux 4.14.85-rancher #1 SMP Sat Dec 1 12:40:08 UTC 2018 x86_64 prometheus-0 (none))","level":"info","ts":"2019-07-02T08:35:21.998Z"}
{"caller":"main.go:325","fd_limits":"(soft=1000000, hard=1000000)","level":"info","ts":"2019-07-02T08:35:21.998Z"}
{"caller":"main.go:326","level":"info","ts":"2019-07-02T08:35:21.998Z","vm_limits":"(soft=unlimited, hard=unlimited)"}
{"caller":"main.go:645","level":"info","msg":"Starting TSDB ...","ts":"2019-07-02T08:35:22.004Z"}
{"address":"0.0.0.0:9090","caller":"web.go:417","component":"web","level":"info","msg":"Start listening for connections","ts":"2019-07-02T08:35:22.005Z"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1560772800000,"mint":1560708000000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.125Z","ulid":"01DDJZG91F6M3SDGHC5KRWN32V"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1560837600000,"mint":1560772800000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.126Z","ulid":"01DDMX4MZFR20672795AE3D28A"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1560902400000,"mint":1560837600000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.127Z","ulid":"01DDPTY41PN8PC94CH92DD2G2C"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1560967200000,"mint":1560902400000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.128Z","ulid":"01DDRRQW571Z3PP4AR0BKDS99K"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561032000000,"mint":1560967200000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.129Z","ulid":"01DDTPHDCF11XTSPF7GN3T2TSP"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561096800000,"mint":1561032000000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.130Z","ulid":"01DDWMB4AY48T33HZJ0VESEBGX"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561161600000,"mint":1561096800000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.132Z","ulid":"01DDYJ4RAEQWZMXXVHZT80BB42"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561226400000,"mint":1561161600000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.133Z","ulid":"01DE0FXW86RFAVVXYK8Q6F1A4G"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561291200000,"mint":1561226400000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.134Z","ulid":"01DE2DQH8WY62QXGM31WH7DK99"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561356000000,"mint":1561291200000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.135Z","ulid":"01DE4BJ04T4T0AT4780PGKHNP1"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561420800000,"mint":1561356000000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.137Z","ulid":"01DE69B6E6T654EW3CVWXEGZ2G"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561485600000,"mint":1561420800000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.138Z","ulid":"01DE876SZZ3XJ6PH4D7J9JE4ZE"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561550400000,"mint":1561485600000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.139Z","ulid":"01DEA50SSGH2F7E0F60D02Y4R1"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561615200000,"mint":1561550400000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.140Z","ulid":"01DEC2TP4MF97SBQZWH1Q8FEK9"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561680000000,"mint":1561615200000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.141Z","ulid":"01DEE0K3ANR0CYQ4Z2QWZSFCVM"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561744800000,"mint":1561680000000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.144Z","ulid":"01DEFYBAH4PHA5QHRW94ABAWVM"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561809600000,"mint":1561744800000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.145Z","ulid":"01DEHW4XDECAX99DPCRPZHFWSE"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561874400000,"mint":1561809600000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.146Z","ulid":"01DEKSYBQWNV36VCC85SCTY0RS"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1561939200000,"mint":1561874400000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.147Z","ulid":"01DENQR0W9YV1P3WZ0SVFQYV90"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1562011200000,"mint":1562004000000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.148Z","ulid":"01DEQNCP5G2MSSFEREJ2AM13YW"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1562004000000,"mint":1561939200000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.149Z","ulid":"01DEQNZDNWXN7GXTCH5XX4ZJJ4"}
{"caller":"repair.go:59","component":"tsdb","level":"info","maxt":1562018400000,"mint":1562011200000,"msg":"found healthy block","ts":"2019-07-02T08:35:22.150Z","ulid":"01DEQW8DDHTERSY8FFNHTPTXVV"}

```

* Pod Description
```
  prometheus:
    Port:          9090/TCP
    Host Port:     0/TCP
    Args:
      --config.file=/etc/prometheus/prometheus.yml
      --storage.tsdb.path=/prometheus
      --storage.tsdb.retention=15d
      --web.enable-lifecycle
      --web.external-url=https://prometheus.xxx
      --log.format=json
      --log.level=debug
    State:          Running
      Started:      Tue, 02 Jul 2019 09:39:23 +0200
    Last State:     Terminated
      Reason:       Error
      Exit Code:    137
      Started:      Tue, 02 Jul 2019 09:28:50 +0200
      Finished:     Tue, 02 Jul 2019 09:39:22 +0200
    Ready:          True
    Restart Count:  2
    Limits:
      cpu:     4
      memory:  64Gi
    Requests:
      cpu:        2
      memory:     16Gi
    Environment:  <none>
    Mounts:
      /etc/prometheus from prometheus-config-emptydir (ro)
      /etc/prometheus/rules from prometheus-rules-emptydir (rw)
      /prometheus from data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from prometheus-token-n2j5l (ro)

```

**Comment by Kirchen99** [comment]
It works after I deleted the PVC.
Is it possible that it is caused by too much metrics? We have more than 200G metrics.

**Comment by krasi-georgiev** [comment]
as commented in the other issue:

> I have not seen any issues with Prometheus that don't lead to some error logs or a panic so you need to dig deeper why is it restarted in he k8s env.

**Comment by krasi-georgiev** [comment]
closing for no activity. Feel free to reopen if you find time to troubleshoot this.

**Comment by pabloli84** [comment]
Maybe it could be helpful for someone.
I had the very same issue in the similar setup and saw the same log.
It happened after increasing number of metrics and adding new service discover configurations.
The issue also was gone after deleting PVC, but then appeared again in some time.

What helped - memory limits increase for Prometheus POD. Though it was not complaining to lack of RAM and wasn't killed by OOM, I noticed that container just restarts every time it reaches RAM limits (I saw it on graphs in monitoring system). So after RAM was increased 3 times in my case, Prometheus started to work fine.
